vs-code-extension/client/src/extension.ts

```diff
@@ -0,0 +1,407 @@
+				for (const [uri, statements] of coveredLines) {
+					coverage.push(
+						vscode.FileCoverage.fromDetails(
+							vscode.Uri.parse(uri),
+							statements.filter((s): s is vscode.StatementCoverage => !!s)
+						)
+					);
+				}
+
+				return coverage;
+			},
+		};
+
+		discoverTests(request.include ?? gatherTestItemsFromCollection(ctrl.items)).then(runTestQueue);
+	};
+
+	ctrl.refreshHandler = async () => {
+		await Promise.all(getWorkspaceTestPatterns().map(({ pattern }) => findRootFiles(ctrl, pattern)));
+	};
+
+	ctrl.createRunProfile('Run Tests', vscode.TestRunProfileKind.Run, runHandler, true, undefined, true);
+
+	ctrl.resolveHandler = async item => {
+		if (!item) {
+			context.subscriptions.push(...startWatchingWorkspace(ctrl, fileChangedEmitter));
+			return;
+		}
+		const data = testData.get(item);
+		if (data instanceof TestFile) {
+			await data.updateFromDisk(ctrl, item);
+		}
+	};
+
+	async function updateNodeForDocument(e: vscode.TextDocument) {
+		if (e.uri.scheme !== 'file') {
+			return;
+		}
+		if (!e.uri.path.endsWith('.cut')) {
+			return;
+		}
+
+		const { file, data } = await getOrCreateFile(ctrl, e.uri);
+		data.updateFromContents(ctrl, e.getText(), file);
+	}
+
+	for (const document of vscode.workspace.textDocuments) {
+		await updateNodeForDocument(document);
+	}
+
+	context.subscriptions.push(
+		vscode.workspace.onDidOpenTextDocument(updateNodeForDocument),
+		vscode.workspace.onDidChangeTextDocument(e => updateNodeForDocument(e.document)),
+	);
+}
+
+export function deactivate() {
+	stopCutLanguageClientServer();
+}
+
+
+async function createDirectoryItems( controller:vscode.TestController, uri: vscode.Uri){
+	// Create TestFile for each directory
+
+	var isInsideTestSuite: boolean = await getIsInsideTestSuiteDirectory1(uri.fsPath)
+	const directory = vscode.workspace.asRelativePath(uri.fsPath)
+	let splitChar = '/'
+	if(!directory.includes(splitChar))
+		splitChar = '\\'
+	const dirArr = directory.split(splitChar);
+	const rootDir = vscode.workspace.getWorkspaceFolder(uri).uri.fsPath;
+	const rootUri = rootDir + '/' + dirArr[0]
+
+	let tmpUri = vscode.Uri.file(rootUri);
+	var file :vscode.TestItem = null; 
+	var data = null
+
+	if(!controller.items.get(rootUri)){ 
+		file = controller.createTestItem(rootUri, dirArr[0], tmpUri);
+		controller.items.add(file);
+		data = new TestFile();
+		data.setDirectoryDetails(rootUri)
+		file.canResolveChildren = true;
+		testData.set(file, data);
+		if(isInsideTestSuite) data.setIsInsideTestSuiteDirectory(true)
+	}
+	else{ 
+		file = controller.items.get(rootUri)
+		data = testData.get(file)
+		if(isInsideTestSuite) data.setIsInsideTestSuiteDirectory(true)
+	}
+	if(dirArr.length==1) return {file,data};
+
+	var prevFile: vscode.TestItem = file
+	var tmpDir = rootUri
+	var tmpFile = null
+	var tmpData = null
+
+	for(var i =1;i<dirArr.length;i++){
+		tmpDir = tmpDir + '/' + dirArr[i]
+		const existing = prevFile.children.get(tmpDir);
+		
+		if(!existing){
```

Comment:
Please reverse this if else statement.
So that we avoid the negation in the if statement.

Reply:
Done